### ISSUE TITLE
Use in JAVA

### ISSUE THREAD

**Issue by rahulshah007** [miso01/SingleRowCalendar#7]
Is there any way to use this library in the JAVA project or I need to use strictly Koitlin for this?

**Comment by MriDx** [comment]
no, it can be implemented in java codes too. I used this library for one of my project with java , android

**Comment by Androstark** [comment]
> no, it can be implemented in java codes too. I used this library for one of my project with java , android

Hi i am also using java but in the bindDataToCalenderView method when using the holder.itemView doesn't find any view to set the text. Can you please help me

**Comment by MriDx** [comment]
Here it worked for me 
![Uploading Screenshot_20200927-165122.jpg…]()

**Comment by MriDx** [comment]
Try like this 
![Uploading Screenshot_20200927-165122.jpg…]()

**Comment by MriDx** [comment]
![Screenshot_20200927-165122](https://user-images.githubusercontent.com/30206346/94363759-34994180-00e2-11eb-8c5a-aa379b6a5e70.jpg)
Try like this

**Comment by MriDx** [comment]
> > no, it can be implemented in java codes too. I used this library for one of my project with java , android
> 
> Hi i am also using java but in the bindDataToCalenderView method when using the holder.itemView doesn't find any view to set the text. Can you please help me

`@Override
            public void bindDataToCalendarView(@NotNull SingleRowCalendarAdapter.CalendarViewHolder calendarViewHolder, @NotNull Date date, int i, boolean b) {
                ((MaterialTextView) calendarViewHolder.itemView.findViewById(R.id.tv_month_calendar_item)).setText(getMonth3Later(date));
                ((MaterialTextView) calendarViewHolder.itemView.findViewById(R.id.tv_date_calendar_item)).setText(DateUtils.INSTANCE.getDayNumber(date));
                ((MaterialTextView) calendarViewHolder.itemView.findViewById(R.id.tv_day_calendar_item)).setText(DateUtils.INSTANCE.getDay3LettersName(date));
            }`

try like this

**Comment by aruns2896** [comment]
@MriDx Not getting the object itemView from calendarViewHolder

How can you able to get that?

**Comment by MamtaSiddhi** [comment]
> no, it can be implemented in java codes too. I used this library for one of my project with java , android

Will you plz explain it how you done it ?

**Comment by MriDx** [comment]
> 

what problem are you facing. ?

**Comment by MriDx** [comment]
> > no, it can be implemented in java codes too. I used this library for one of my project with java , android
> 
> Will you plz explain it how you done it ?

What problem are you facing ?

**Comment by MamtaSiddhi** [comment]
Same unable to get ItemView in java code implemented sdk of it 

> > > no, it can be implemented in java codes too. I used this library for one of my project with java , android
> > 
> > 
> > Will you plz explain it how you done it ?
> 
> What problem are you facing ?

Same unable to get ItemView in java code implemented sdk of it

**Comment by MriDx** [comment]
@MamtaSiddhi 
You won't get the itemview directly, you will receive viewholder of type "CalendarViewHolder".

**Comment by MamtaSiddhi** [comment]
> 

Tried but not get it 

calendarViewHolder.getItemDetails

getting this 

not View

**Comment by MriDx** [comment]
> 



> > 
> 
> Tried but not get it
> 
> calendarViewHolder.getItemDetails
> 
> getting this
> 
> not View

View itemView = calendarViewHolder.itemView;

itemView.findViewById(R.id.viewId).setText("hello world");

access it like this

**Comment by sahil1772** [comment]
> > 
> 
> > > 
> > 
> > 
> > Tried but not get it
> > calendarViewHolder.getItemDetails
> > getting this
> > not View
> 
> View itemView = calendarViewHolder.itemView;
> 
> itemView.findViewById(R.id.viewId).setText("hello world");
> 
> access it like this

Sorry but it returns getItemDetails() no matter how you accept the data. The itemView object is never returned. How do you handle the getItemDetails to receive at least the Id of the Views in it?
